hypothesis-python/tests/array_api/common.py

```diff
@@ -32,11 +38,12 @@
 # which should allow our test suite to still work. A constant is set accordingly
 # to inform our test suite of whether the array module here is a mock or not.
 try:
-    with pytest.warns(UserWarning):
-        from numpy import array_api as xp  # type: ignore
+    with catch_warnings():  # libraries might warn on importing their namespace
+        modules = installed_array_modules()
+    xp = modules["numpy"]
     xps = make_strategies_namespace(xp)
     COMPLIANT_XP = True
-except ImportError:
+except KeyError:
     xp = mock_xp
     with pytest.warns(HypothesisWarning):
         xps = make_strategies_namespace(xp)
```

Comment:
```python
modules = installed_array_modules()
try:
    xp = modules["numpy"].load()
except KeyError:
    xp = mock_xp
    with pytest.warns(HypothesisWarning):
        xps = make_strategies_namespace(xp)
    COMPLIANT_XP = False
else:
    xps = make_strategies_namespace(xp)
    COMPLIANT_XP = True
```

Tighter scoping for the exception handling, and relies on the deferred-`.load()` mentioned above.